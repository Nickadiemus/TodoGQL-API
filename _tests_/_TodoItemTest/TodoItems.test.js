const { FetchTodoItems, FetchSingleTodoItem } = require('./TodoItems');
const id = 'mb4zcsEbGwzA6XuTz';


/*/
 *  Type of Test: Query Snapshot
 *  Proves: graphql Endpoint returns actual data equal to a snapshot
 *  @param(s): none
 *  Description: an asynchronous call to the localhost:3000/graphql Endpoint
 *  that will verify by a snapshot the query matches the response given

 NOTE: TodoItems must be populated before this test can be ran
/*/


test('test call FetchTodoItems', async () => {
  const data = await FetchTodoItems();
  console.log(data);
  expect(data).toMatchSnapshot();
})

/*/
 *  Type of Test: Query Snapshot
 *  Proves: graphql Endpoint returns actual data equal to a snapshot
 *  @param(s): id
 *  Description: an asynchronous call to the localhost:3000/graphql Endpoint
 *  that will verify a single TodoItem by a snapshot the query matches the response
 *  given

 NOTE: TodoItems must be populated before this test can be ran and an id must
 be provided
/*/

test('test call for FetchSingleTodoItem(id)', async () => {
  const data = await FetchSingleTodoItem(id);
  expect(data).toMatchSnapshot();
})

/*/
 *  Type of Test: Unit Testing
 *  Proves: data returned from API are corresponding to defined schemas
 *  @param(s): none
 *  Description: an asynchronous call to the localhost:3000/graphql Endpoint
 *  that will unit test every expected unit of TodoItem

 NOTE: TodoItems must be populated before this test can be ran
/*/


it('Unit Test for all TodoItems on Request', async () => {
  const data = await FetchTodoItems();
  count = 0;
  data.data.TodoItems.forEach(item => {
    expect(typeof(item._id)).toBe('string');
    expect(typeof(item.parent)).toBe('string');
    expect(typeof(item.content)).toBe('string');
    expect(typeof(item.completed)).toBe('boolean');
    count++;
  })
  console.log("Loop(s): " + count);
})
