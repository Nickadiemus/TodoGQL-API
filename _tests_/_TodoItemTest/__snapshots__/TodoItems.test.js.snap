// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test call FetchTodoItems 1`] = `
Object {
  "data": Object {
    "TodoItems": Array [
      Object {
        "_id": "mb4zcsEbGwzA6XuTz",
        "completed": false,
        "content": "TodoItem1",
        "parent": "TtqWyeRJLbC67Hs7G",
      },
      Object {
        "_id": "xavRNCd5RJfhTdJjK",
        "completed": false,
        "content": "TodoItem2",
        "parent": "TtqWyeRJLbC67Hs7G",
      },
      Object {
        "_id": "7W6dsBuZpRP7233mE",
        "completed": false,
        "content": "TestItem1",
        "parent": "DeFkpyTASdvxGnSmX",
      },
      Object {
        "_id": "vXMt2uzo8rbheR7ju",
        "completed": false,
        "content": "TestItem2",
        "parent": "DeFkpyTASdvxGnSmX",
      },
      Object {
        "_id": "bkgAg6PbHeeGGZnpZ",
        "completed": false,
        "content": "TestItem3",
        "parent": "DeFkpyTASdvxGnSmX",
      },
      Object {
        "_id": "GFsG62dTm2MYob6d3",
        "completed": false,
        "content": "TestItem1",
        "parent": "shQMsTn2uHFzAYBmg",
      },
    ],
  },
}
`;

exports[`test call for FetchSingleTodoItem(id) 1`] = `
Object {
  "data": Object {
    "FindTodoItem": Object {
      "_id": "mb4zcsEbGwzA6XuTz",
      "completed": false,
      "content": "TodoItem1",
      "parent": "TtqWyeRJLbC67Hs7G",
    },
  },
}
`;
